### PR TITLE
Fix GCC implicit-fallthrough= warnings

### PR DESCRIPTION
**Pull request by ozone10**
fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13142

### CODE REVIEW

File: PowerEditor/src/NppBigSwitch.cpp [hunk 1]
```diff
@@ -3290,6 +3293,7 @@ LRESULT Notepad_plus::process(HWND hwnd, UINT message, WPARAM wParam, LPARAM lPa
 			{
 				addHotSpot(pView);
 			}
+			return TRUE;
```
Comment (donho) [inline PowerEditor/src/NppBigSwitch.cpp:3296]: 👍

File: PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp [hunk 2]
```diff
@@ -1520,6 +1520,8 @@ intptr_t CALLBACK UserDefineDialog::run_dlgProc(UINT message, WPARAM wParam, LPA
             }
             ::SetScrollPos(_hSelf, SB_VERT, _yScrollPos, TRUE);
             ::ScrollWindow(_hSelf, 0, oldy-_yScrollPos, NULL, NULL);
+
+            break;
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp:1524]: 👍

File: PowerEditor/src/ScintillaComponent/columnEditor.cpp [hunk 3]
```diff
@@ -436,8 +436,10 @@ intptr_t CALLBACK ColumnEditorDlg::run_dlgProc(UINT message, WPARAM wParam, LPAR
 						}
 						break;
 					}
+					break;
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/columnEditor.cpp:439]: 👍

File: PowerEditor/src/WinControls/AboutDlg/AboutDlg.cpp [hunk 4]
```diff
@@ -390,6 +391,7 @@ intptr_t CALLBACK DebugInfoDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM
 				default:
 					break;
 			}
+			break;
```
Comment (donho) [inline PowerEditor/src/WinControls/AboutDlg/AboutDlg.cpp:394]: 👍

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 5]
```diff
@@ -1458,6 +1458,8 @@ intptr_t CALLBACK DarkModeSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPARA
 			delete _pLinkColorPicker;
 			delete _pHotEdgeColorPicker;
 			delete _pDisabledEdgeColorPicker;
+
+			return TRUE;
```
Comment (donho) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:1462]: 👍

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 6]
```diff
@@ -2531,6 +2536,8 @@ intptr_t CALLBACK DefaultDirectorySubDlg::run_dlgProc(UINT message, WPARAM wPara
 			::EnableWindow(::GetDlgItem(_hSelf, IDD_OPENSAVEDIR_ALWAYSON_BROWSE_BUTTON), shouldActivated);
 
 			::SendDlgItemMessage(_hSelf, IDC_OPENSAVEDIR_CHECK_DRROPFOLDEROPENFILES, BM_SETCHECK, nppGUI._isFolderDroppedOpenFiles ? BST_CHECKED : BST_UNCHECKED, 0);
+
+			return TRUE;
```
Comment (donho) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:2540]: 👍

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 7]
```diff
@@ -2662,6 +2669,8 @@ intptr_t CALLBACK RecentFilesHistorySubDlg::run_dlgProc(UINT message, WPARAM wPa
 			::SetDlgItemInt(_hSelf, IDC_CUSTOMIZELENGTHVAL_STATIC, length, FALSE);
 			_customLenVal.init(_hInst, _hSelf);
 			_customLenVal.create(::GetDlgItem(_hSelf, IDC_CUSTOMIZELENGTHVAL_STATIC), nullptr);
+
+			return TRUE;
```
Comment (donho) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:2673]: 👍

**Comment by donho** [comment]
@ozone10 
It's a good warning, however it seems not possible to enable it while compiling with MSVC.
The only way I found is to use **C++ Core Guidelines checkers** to detect it.
If you know how to enable it with MSVC, feel free to add it into the project file in the future.

**Comment by ozone10** [comment]
@donho 
> It's a good warning, however it seems not possible to enable it while compiling with MSVC. The only way I found is to use **C++ Core Guidelines checkers** to detect it. If you know how to enable it with MSVC, feel free to add it into the project file in the future.

Using custom rule set and code analysis should allow to enable it. I will see what I can do.
Take a note that using code analysis will significantly increase compiling time.

**Pull request closed without merging**